spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/PublisherCallbackChannelImpl.java

```diff
@@ -119,7 +121,8 @@ public PublisherCallbackChannelImpl(Channel delegate) {
 	public PublisherCallbackChannelImpl(Channel delegate, @Nullable ExecutorService executor) {
 		delegate.addShutdownListener(this);
 		this.delegate = delegate;
-		this.executor = executor != null ? executor : DEFAULT_EXECUTOR;
+		this.executor = executor != null ? executor : Executors.newSingleThreadExecutor();
```

Comment:
No; we don't want a new executor for every channel; that's why it's static.

I suggest we move the static executor to the CCF (non-static there though) and always pass it into the ctor; then shut down the executor in the `CCF.destroy()`.